
<launch>

    <!-- Start a PID controller to manage the robot rotation-->
    <remap from="rotation_state" to="/user_rotation/state"/>
    <remap from="rotation_setpoint" to="/user_rotation/setpoint"/>
    <remap from="user_effort" to="/user_rotation/control_effort"/>
    
    <node name="user_rotation_pid" pkg="pid" type="controller" ns="user_rotation">
          <param name="Kp" value="0.001" />
          <param name="Ki" value="0.0001" />
          <param name="Kd" value="0.0" />
    </node>

    <!-- Launch person following robot image analysis node-->
    <node name="image_analyser" pkg="pfr_package" type="pfr_image_evaluation.py" output="screen" />

    <!-- Start Base Controller launch file -->
    <include file="$(find pfr_package)/launch/base_control.launch" />

    <!-- Start Motion Controller-->
    <node name="motion_controller" pkg="pfr_package" type="motion_controller.py" output="screen" />

    <!-- Start Button state publisher-->
    <node name="button_publisher" pkg="pfr_package" type="button_state_pub.py" output="log" />
    
</launch>
